brave/src/main/java/brave/internal/Platform.java

```diff
@@ -76,18 +77,27 @@ public static Platform get() {
 
   /** Like {@link Logger#log(Level, String) */
   public void log(String msg, @Nullable Throwable thrown) {
-    if (!LOG.isLoggable(Level.FINE)) return; // fine level to not fill logs
-    LOG.log(Level.FINE, msg, thrown);
+    Logger logger = logger();
+    if (!logger.isLoggable(Level.FINE)) return; // fine level to not fill logs
+    logger.log(Level.FINE, msg, thrown);
   }
 
   /** Like {@link Logger#log(Level, String, Object)}, except with a throwable arg */
   public void log(String msg, Object param1, @Nullable Throwable thrown) {
-    if (!LOG.isLoggable(Level.FINE)) return; // fine level to not fill logs
+    Logger logger = logger();
+    if (!logger.isLoggable(Level.FINE)) return; // fine level to not fill logs
     LogRecord lr = new LogRecord(Level.FINE, msg);
     Object[] params = {param1};
     lr.setParameters(params);
     if (thrown != null) lr.setThrown(thrown);
-    LOG.log(lr);
+    logger.log(lr);
+  }
+
+  Logger logger() {
+    Logger result = this.logger;
+    if (result != null) return result;
+    // synchronization not needed as worst case we overwrite the field twice
```

Comment:
I think worst case is overwriting it 1000 times with 1000 threads (very unlikely of course). But anyways, if not switching to nested class pattern, I'd make the field non-volatile since we're doing best-effort only to minimize logger instantiation anyways.